addons/stock/models/stock_move.py

```diff
@@ -598,6 +598,13 @@ def write(self, vals):
                 receipt_moves_to_reassign |= move_to_unreserve.filtered(lambda m: m.location_id.usage == 'supplier')
                 receipt_moves_to_reassign |= (self - move_to_unreserve).filtered(lambda m: m.location_id.usage == 'supplier' and m.state in ('partially_available', 'assigned'))
                 move_to_recompute_state |= self - move_to_unreserve - receipt_moves_to_reassign
+        # propagate packaging_id changes in the stock move chain
+        if 'product_packaging_id' in vals:
+            if self.move_dest_ids and \
```

Comment:
Updating move_dest_ids is relevant only in the reception flows, but useless in delivery flows (in that case you need to propagate on move_orig_ids).
Doing it may result in recursion issue (writing on move_dest_ids then move_orig_ids, etc) so you could do the same we did in `_set_date_deadline()` with the `already_propagated_ids` key